File: ProgrammingProblems/2017/darts.java
import java.util.Scanner;

public class darts {
	public static void main(String[] args) {
		try (Scanner in = new Scanner(System.in)) {
			int cases = in.nextInt();
			for (int a = 0; a < cases; a++) {
				int w = in.nextInt(), b = in.nextInt(), d = in.nextInt(), s = in.nextInt();
				
				int throwCount = in.nextInt();
				int score = 0;
				for (int i = 0; i < throwCount; i++){
					double x = in.nextDouble(), y = in.nextDouble();
					double r = Math.sqrt(x*x+y*y);
					double theta = Math.atan2(y, x);
					if (theta < 0){
						theta += 2.0*Math.PI;
					}
					int wedge = (int)(theta/(2.0*Math.PI)*w)+1;
					
					if (r < b){
						score+=50;
					}
					else if(r < d){
						score += wedge*2;
					}else if(r < s){
						score += wedge;
					}
					
				}
				System.out.println(score);
			}
		}
	}
}

File: ProgrammingProblems/2017/darts.py
import math
cases = int(input())
for a in range(cases):
    [w,b,d,s] = map(lambda b : int(b), input().split(" "))
    throwCount = int(input())
    score = 0
    for i in range(throwCount):
        [x,y] = map(lambda b : float(b), input().split(" "))
        r = math.hypot(x,y)
        theta = math.atan2(y,x)
        if (theta < 0):
            theta+=2*math.pi
        throwScore = int(theta/(2*math.pi)*w)+1
        if (r<b):
            score+=50
        elif (r<d):
            score+=throwScore*2
        elif (r<s):
            score+=throwScore
    print(score)